Task 1

AleksandrParamonov opened the pull request.

AleksandrParamonov committed:
Beta release

First Github release.
Everythink except dump is working. 5/5 test are passed

AleksandrParamonov committed:
Removed elder version backup

No more rubbish

igorsmir-ilab commented on func_sim/func_memory/func_memory.cpp:
```diff
@@ -11,39 +11,148 @@
 
 FuncMemory::FuncMemory( const char* executable_file_name,
-                        uint64 addr_size,
+                        uint64 addr_size_global,
```
I suggest you not changing default names of variables

igorsmir-ilab commented on func_sim/func_memory/func_memory.cpp:
```diff
@@ -11,39 +11,148 @@
 {
-    // put your code here
-    return string("ERROR: You need to implement FuncMemory!");
+    return string("Not yet!");
```
I recommend you to look at ELF parser _dump()_ method if you face difficulties with its realization.

AleksandrParamonov replied:
Hmm, I used REPLACE in editor to rename local variables to make it fit
the guidelines, and somehow this replace affected default variable name. I
will fix it ASAP.

2015-10-23 17:43 GMT+03:00 Igor Smirnov notifications@github.com:

> In func_sim/func_memory/func_memory.cpp
> https://github.com/MIPT-ILab/mipt-mips-2015/pull/5#discussion_r42872714:
> 
> >  }
> > 
> >  string FuncMemory::dump( string indent) const
> >  {
> > -    // put your code here
> > -    return string("ERROR: You need to implement FuncMemory!");
> > -    return string("Not yet!");
> 
> I recommend you to look at ELF parser _dump()_ method if you face
> difficulties with its realization.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/MIPT-ILab/mipt-mips-2015/pull/5/files#r42872714.

AleksandrParamonov replied:
https://github.com/AleksandrParamonov/mipt-mips-2015/commit/1573ff73a424cfbeff56a09c7942647b897afd9c
Issues are fixed.

2015-10-23 18:30 GMT+03:00 Александр Сергеевич Парамонов <
aleksandr.paramonov@phystech.edu>:

> Hmm, I used REPLACE in editor to rename local variables to make it fit
> the guidelines, and somehow this replace affected default variable name. I
> will fix it ASAP.
> 
> 2015-10-23 17:43 GMT+03:00 Igor Smirnov notifications@github.com:
> 
> > In func_sim/func_memory/func_memory.cpp
> > https://github.com/MIPT-ILab/mipt-mips-2015/pull/5#discussion_r42872714
> > :
> > 
> > >  }
> > > 
> > >  string FuncMemory::dump( string indent) const
> > >  {
> > > -    // put your code here
> > > -    return string("ERROR: You need to implement FuncMemory!");
> > > -    return string("Not yet!");
> > 
> > I recommend you to look at ELF parser _dump()_ method if you face
> > difficulties with its realization.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/MIPT-ILab/mipt-mips-2015/pull/5/files#r42872714.

AleksandrParamonov committed:
Added dump function

- dump is working now;
-"changing default names of variables" bug is fixed;
- no more enormous lines
